### ISSUE TITLE
[ExpoKit] Sentry Logger [Error]: Error while sending event: [object Object]

### ISSUE THREAD

**Issue by otaviogaiao** [expo/sentry-expo#90]
Hello, I'm having an issue after upgrading sentry to 2.0.1 on my expoKit app with SDK 35.
When I try to use Sentry.captureException, I get an error and the app reloads. This is happening both in production and in dev mode when on. Does anyone have any idea about what it can be?

Here are my configurations:

**App.js:**
```
Sentry.init({
  dsn: '------',
  enableInExpoDevelopment: true,
  debug: true,
});
```

**app.json**
```
    "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "sentry",
            "project": "....",
            "authToken": ".....",
            "url": "https://..."
          }
        }
      ]
    }
```

Every time I call `Sentry.captureException(e);`, I get:

<img width="910" alt="Screen Shot 2019-11-19 at 19 38 45" src="https://user-images.githubusercontent.com/3965485/69192854-37abc480-0b04-11ea-8e2f-dead6c35956e.png">

**Comment by HazAT** [comment]
What is `e` in you `Sentry.captureException(e);` call?

**Comment by otaviogaiao** [comment]
> What is `e` in you `Sentry.captureException(e);` call?

It’s an error object, usually thrown by a failed network request. It worked before I updated to sentry 2.0.

**Comment by cruzach** [comment]
@otaviogaiao if you `captureException('random text string')`, do you see the same error?

**Comment by otaviogaiao** [comment]
> @otaviogaiao if you `captureException('random text string')`, do you see the same error?

Yes I do

**Comment by cruzach** [comment]
Hm, tested that on `sentry-expo` 2.0.1 and didn't get any error. Maybe this has something to do with your self-hosting config?

**Comment by laurpantelimon** [comment]
The same happened to me. I don't have the "url" field in config. All I did is calling the .init and .setRelease functions and the Sentry throws this same error. Not instant, when random browsing through app. I also have sentry-expo 2.0.1 and expo 35.

**Comment by cruzach** [comment]
@laurpantelimon you're saying this happens without an error being thrown?

**Comment by laurpantelimon** [comment]
The error is thrown by react native, the red page with the "Sentry Logger [Error]: Error while sending event: [object Object]" message and there is no error thrown on the Sentry web project.

**Comment by cruzach** [comment]
For messages to be logged in development, you need to set `enableInExpoDevelopment: true,` when you initialize. Please share a method for me to reproduce this, otherwise it's hard to say what could cause it

**Comment by laurpantelimon** [comment]
This were the only two Sentry functions in my code 
Sentry.init({
    dsn: SentryDSN,
    enableInExpoDevelopment: true,
    debug: true
});
Sentry.setRelease(Constants.manifest.revisionId);

**Comment by otaviogaiao** [comment]
> This were the only two Sentry functions in my code, nothing else.
> Sentry.init({
> dsn: Se

> For messages to be logged in development, you need to set `enableInExpoDevelopment: true,` when you initialize. Please share a method for me to reproduce this, otherwise it's hard to say what could cause it

The issue is happening on production for me.

**Comment by cruzach** [comment]
tried to repro this on a blank `expo init` project just using the same functions as @laurpantelimon said, still haven't had any issues
```
import React from "react";
import { StyleSheet, Text, View, Button, Image } from "react-native";
import * as Sentry from "sentry-expo";
import Constants from "expo-constasnts";

Sentry.init({
  dsn: "YOUR_DSN_HERE",
  enableInExpoDevelopment: true,
  debug: true
});

Sentry.setRelease(Constants.manifest.revisionId);

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <Button
        title="throw"
        onPress={() => {
          throw new Error("hello error message");
        }}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  }
});
```

**Comment by otaviogaiao** [comment]
> tried to repro this on a blank `expo init` project just using the same functions as @laurpantelimon said, still haven't had any issues
> 
> ```
> import React from "react";
> import { StyleSheet, Text, View, Button, Image } from "react-native";
> import * as Sentry from "sentry-expo";
> import Constants from "expo-constants";
> 
> Sentry.init({
>   dsn: "YOUR_DSN_HERE",
>   enableInExpoDevelopment: true,
>   debug: true
> });
> 
> Sentry.setRelease(Constants.manifest.revisionId);
> 
> export default function App() {
>   return (
>     <View style={styles.container}>
>       <Text>Open up App.js to start working on your app!</Text>
>       <Button
>         title="throw"
>         onPress={() => {
>           throw new Error("hello error message");
>         }}
>       />
>     </View>
>   );
> }
> 
> const styles = StyleSheet.create({
>   container: {
>     flex: 1,
>     backgroundColor: "#fff",
>     alignItems: "center",
>     justifyContent: "center"
>   }
> });
> ```

Are you using it on an ejected project ? I’m only having it on Android in an ejected projected that used sdk 35

**Comment by cruzach** [comment]
ejected to bare or to expokit?

**Comment by otaviogaiao** [comment]
> ejected to bare or to expokit?

Expokit

**Comment by boraerbasoglu** [comment]
same here

**Comment by ludwig-adoria** [comment]
🙋🏻I have the same issue, my project isn’t ejected !

**Comment by ludwig-pro** [comment]
about my package.json 
```
"expo": "^35.0.0",
"sentry-expo": "^2.0.1",
```

**Comment by chad-vc** [comment]
I am seeing the same against our internal instance of Sentry. However I signed up for the trial of hosted Sentry to test that and it worked just fine. I noticed a few API differences between our older installed version of Sentry and the online hosted version, so maybe this library doesn't work with older versions of Sentry?

**Comment by ludwig-adoria** [comment]
yep maybe we just need to upgrade sentry-react-native dep to 1.3.6 version

**Comment by tamdao** [comment]
Any luck with this issue?

**Comment by cruzach** [comment]
@ludwig-adoria currently this repo uses `@sentry/react-native@^1.0.0`. What makes you say that upgrading to 1.3.6 would fix this issue? 

If this is only happening in Expokit, I should note that Expokit has been deprecated since with [SDK 37](https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6), all features available in Expokit are now available (and better!) in the bare workflow

**Comment by cruzach** [comment]
I'm going to close this issue since Expokit is now officially deprecated. If you were using Expokit, you should migrate to the bare workflow now, and we've published a new major version of `sentry-expo` as `next` (3.0.0-rc1) which has much better support for the bare workflow

Definitely try it out, and open a new issue if you run into any problems with it so we can address them before publishing as `latest`. 

Instructions for additional setup if you're using the bare workflow can be found [here](https://github.com/expo/sentry-expo#bare-workflow-setup)

If you had this issue in an un-ejected project, please open a new issue with a minimal reproducible demo
